apps/check-processing-time/src/check_processing_time.py

```diff
@@ -0,0 +1,14 @@
+import json
+
+
+def get_time_from_attempts(attempts):
+    attempts.sort(key=lambda attempt: attempt['StartedAt'])
+    final_attempt = attempts[-1]
+    return final_attempt['StoppedAt'] - final_attempt['StartedAt']
+
+
+def lambda_handler(event, context):
+    results = event['processing_results']
+    if 'Attempts' in results:
+        return get_time_from_attempts(results['Attempts'])
+    return get_time_from_attempts(json.loads(results['Cause'])['Attempts'])
```

Comment:
Not blocking, but could separate the parsing of attempts from the processing to make this a bit more expressive:
```suggestion
    if 'Attempts' in results:
        attempts = results['Attempts']
    else:
        attempts = json.loads(results['Cause'])['Attempts']

    return get_time_from_attempts(attempts)
```